### DISCUSSION TITLE
Get a pdf file in got.post()

### DISCUSSION THREAD

**Discussion by ainaaad** [sindresorhus/got#2012] [Q&A] [answered]
Is it possible with got.post() to receive a fitxero pdf?

I am trying it with this:
       _const res = await got.post(url {
         header,
         bodysuit: bodysuit,
         pdf: true
       });_

But the response gives an error because I receive the response as JSON (_ParseError: Unexpected token % in JSON_ )

**Comment by sindresorhus** [comment, marked as answer]
`pdf: true` isnt a valid got option. to download a PDF (or any binary file), you need to use buffer response type:

```javascript
const pdfBuffer = await got.post(url, {
  headers: header,
  json: bodysuit
}).buffer();

fs.writeFileSync('file.pdf', pdfBuffer);
```

the JSON parsing error happens because by default got tries to parse responses as JSON when you use `.json()` on it, but PDFs are binary data. `responseType: 'buffer'` tells got to return the raw binary data as a Buffer instead.

also, i notice you have `bodysuit` - if thats a typo and you meant to send JSON data, use the `json` option (shown above). if its form data, use `form` instead.
